docs/user_guide/representing/array.md

```diff
@@ -439,6 +439,104 @@ assert not my_doc.is_view()  # False
     To summarize: you should use `DocVec` when you need to work with contiguous data, and you should use `DocList` when you need to rearrange
     or extend your data.
 
+
+## Dealing with Optional Field
+
+
+Both [`DocList`][docarray.array.doc_list.doc_list.DocList] and [`DocVec`][docarray.array.doc_vec.doc_vec.DocVec] support optional fields but they behave differently.
+
+!!! note
+    This whole section is specific for nested BaseDoc, DocList and DocVec. 
+    for other data type DocList and DocVec will treat the optional case as a normal case.
+
+Let's take an example:
+
+```python
+from typing import Optional
+from docarray.typing import NdArray
+import numpy as np
+
+
+class ImageDoc(BaseDoc):
+    tensor: NdArray
+
+
+class ArticleDoc(BaseDoc):
+    image: Optional[ImageDoc]
+    title: str
+```
+
+In this example `ArticleDoc` has an optional field `image` which is an `ImageDoc`. This means that this field can either
+be None or be a `ImageDoc` instance.
+
+Remember for both DocList and DocVec calling `docs.image` will return a list like object of all the images of the documents.
+
+For DocList it will just iterate over all the documents and collect the image attribute of each document in a sequence for DocVec it will return the column of the image attribute.
+
+The question which kind of sequence to you pick when the field is optional, i.e, some of the datapoint could be None ?
+
+For DocList it will return a list of `Optional[ImageDoc]` instead of a `DocList[ImageDoc]`:
+
+```python
+from docarray import DocList
+
+
+docs = DocList[ArticleDoc](
+    [
+        ArticleDoc(image=ImageDoc(tensor=np.ones((3, 224, 224))), title="Hello"),
+        ArticleDoc(image=None, title="World"),
+    ]
+)
+
+assert docs.image == [ImageDoc(tensor=np.ones((3, 224, 224))), None]
+```
+
+but for DocVec it is a bit different. Indeed, DocVec store the data for each filed as contiguous column. 
+This means that DocVec can create a column in only two case: either all the data for a field is None or all the data is not None.
```

Comment:
```suggestion
This means that DocVec can create a column in only two cases: either all the data for a field is None or all the data is not None.
```
I think in other places you have a name for this. Homogeneous? something like that? Would be good to introduce here I think.
Or did we take homogeneous to mean same doc type for every doc? Honestly I don't remember